### PR TITLE
DOC: Add citation info

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #278 

Adds a page to documentation with discussion of citations, including sample bibtex info.

## Type of change

- This change is a documentation update

# How Has This Been Tested?

No changes to code.
# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by jklenzing** [comment]
There's some strangeness transpiring at coveralls.  The number of files seems to double with each additional run.  Up to 332 for this 83 file project.

https://coveralls.io/github/rstoneback/pysat?branch=doc_how_to_cite

**Comment by aburrell** [comment]
@jklenzing I am not sure I am the right reviewer for this, given that I made a few changes.

**Comment by jklenzing** [comment]
OK.  @rstoneback?

File: docs/citing.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    issn      = {21699402},
+    journal   = {Journal of Geophysical Research: Space Physics},
+    number    = {6},
+    pages     = {5271--5283},
+    title     = {{PYSAT: Python Satellite Data Analysis Toolkit}},
+    volume    = {123},
+    year      = {2018}
+  }
+
+To aid in scientific reproducibility, please include the version number in publications that use this code.  This can be found by invoking `pysat.__version__ `.
```
Comment (rstoneback) [inline docs/citing.rst]: The DOI in the citation above points specifically to v2.0 and includes v2.0 in the title. Do we really need this note?

Comment (jklenzing) [reply docs/citing.rst]: Actually, this is something we should discuss.  Are these citations considered examples?  Are we planning to update them with every version?  Either way, we should clarify this area.

**Comment by rstoneback** [comment]
Yeah, let’s talk about it. 

Didn’t we also talk about adding a .cite() method? Perhaps in Japan? We could have one for pysat and and potentially one defined by each instrument.

**Comment by jklenzing** [comment]
Instrument specific citations is mentioned in #180.

**Pull request merged**